test/Sema/diag_non_ephemeral_swift4.swift

```diff
@@ -21,46 +21,46 @@ var optionalArr: [Int8]?
 // We cannot use array-to-pointer and string-to-pointer conversions with
 // non-ephemeral parameters.
 
-takesMutableRaw(&arr, 5) // expected-warning {{passing temporary pointer argument of type 'UnsafeMutableRawPointer' to parameter expecting pointer that outlives the duration of the call leads to undefined behaviour; this will be an error in a future release}}
-// expected-note@-1 {{implicit argument conversion from '[Int8]' to 'UnsafeMutableRawPointer' produces a pointer valid only for the duration of the call}}
+takesMutableRaw(&arr, 5) // expected-warning {{inout expression creates a temporary pointer, but argument #1 should be a pointer that outlives the call to 'takesMutableRaw'}}
```

Comment:
It's weird to have a diag_non_ephemeral_swift4.swift without the corresponding diag_non_ephemeral_swift5.swift.

Reply:
Yeah, that's a holdover from when I had the logic combined for a warning in Swift 4 mode and error for Swift 5 mode. I'll rename it to `diag_non_ephemeral_warning`.

Thanks for the review!